### ISSUE TITLE
Document how we update our docs

### ISSUE THREAD

**Issue by gregkalapos** [elastic/apm-agent-dotnet#1842]
Currently [our doc page](https://www.elastic.co/guide/en/apm/agent/dotnet/current/index.html) is out-of-sync. `current` there is `1.12`, but the latest release is `1.17.0` (which should be `current`)

The way the docs are currently updated is that we make a PR in https://github.com/elastic/docs - an example PR: https://github.com/elastic/docs/pull/2154. 

Most APM Agent use a different approach: they set a specific branch in the APM Agent repo as current (let's say `1.x`) in the docs repo and for each agent release they just update their branch in the agent repo (in this case the `1.x` branch) which will then include the current documentation. 

The .NET Agent tries to be aligned on how docs are versioned with the overall stack docs, so on each minor (actually patch) release we need to update the docs repo as well. More on this in this PR: https://github.com/elastic/docs/pull/2066

Things which aren't really clear and I'd like to clarify here:
- [x] What files do we need to touch in https://github.com/elastic/docs when we release a new .NET agent version? You can see a few PRs here: https://github.com/elastic/docs/pulls?q=is%3Apr+dotnet+is%3Aclosed, which do the same changes across multiple files (I assume this has something to do with maintaining different versions). But it's not clear what files we need to change. If we figure this out, I'll document it in https://github.com/elastic/apm-agent-dotnet/blob/main/RELEASING.md. I'd like to have some clear rules around it, like: "If we release a new patch/minor/major .NET Agent version, then we need to update files X,Y,X (or some rule probably related to supported versions) in the docs repo"... or something similar, which we will then follow each time we release.
- [x] Once it's clear which files we need to update, I'll open a PR to make our docs up-to-date.

@elastic/obs-docs @bmorelli25 Could you please help me to figure out the 1. point above? Happy to jump on zoom if needed.

**Comment by bmorelli25** [comment]
Hey Greg 👋.

There are two locations where updates need to be made in [elastic/docs](https://github.com/elastic/docs). The first is a single file, `conf.yaml`. And the second is a directory of files, `shared/versions/stack/*.asciidoc`.

#### [`conf.yaml`](https://github.com/elastic/docs/blob/master/conf.yaml)

This file is the config file for our documentation build. It specifies the sources of each documentation book, the titles of the books, the versions to publish, etc. The three lines you're interested in are [here](https://github.com/elastic/docs/blob/16751d2c5ccce4817d90bdc4110c41458c322c43/conf.yaml#L1332-L1334).

1. `current` — specifies the current version of the documentation. This version must match a branch name in the `apm-agent-dotnet` repo.
2. `branches` — the names of branches in the `apm-agent-dotnet` repo that you wish to build documentation for.
3. `live` — the names of the branches that show up in the version dropdown. Any branches not specified here, but are specified in the `branches` list are visible by clicking "other versions" in the version selector dropdown.

<img width="316" alt="Screen Shot 2022-09-27 at 1 04 35 PM" src="https://user-images.githubusercontent.com/5618806/192624619-0c228fff-be72-4377-81f0-ac759f511bcc.png">

When a new version of the agent is released, all three lines above need to be updated. If you're updating to 1.17, the change would look something like this:

```diff
-                    current:    1.12
-                    branches:   [ {main: master}, 1.12, 1.11, 1.10, 1.9, 1.8 ]
-                    live:       [ main, 1.12 ]
+                    current:    1.17
+                    branches:   [ {main: master}, 1.16, 1.15, 1.14, 1.13, 1.12, 1.11, 1.10, 1.9, 1.8 ]
+                    live:       [ main, 1.17 ]
```

#### [`shared/versions/stack/*.asciidoc`](https://github.com/elastic/docs/tree/master/shared/versions/stack)

This directory defines how links from stack versioned documentation relate to links from non-stacked versioned documentation. For example, in the 8.5 file, the variable `:apm-dotnet-branch:` is set to `1.12`. This means any links in the 8.5 stack docs (like the APM Guide) that point to the APM Agent .NET Reference, will point to the 1.12 version of those docs.

https://github.com/elastic/docs/blob/16751d2c5ccce4817d90bdc4110c41458c322c43/shared/versions/stack/8.5.asciidoc#L44

The number of files you update in this directory depends on version compatibility between stack docs and your APM agent. In general, we update as far back as the new version of the agent is compatible with the Stack; this pushes new documentation to the user.

---

Let me know if I can explain further. Happy to jump on a call too—feel free to throw something on my calendar!

**Comment by bmorelli25** [comment]
I posted in another PR, but let me know if you'd like me to open a PR to update the docs to current.

**Comment by bmorelli25** [comment]
Hey, @gregkalapos. I've opened https://github.com/elastic/docs/pull/2538 to add newer versions to the docs, but I'm running into an error: `12:59:08 INFO:build_docs:Current branch <1.17> is not in <branches> in book <APM .NET Agent> at lib/ES/Book.pm line 118.`

I only see branches through 1.12:
<img width="732" alt="Screen Shot 2022-10-06 at 1 44 26 PM" src="https://user-images.githubusercontent.com/5618806/194415007-139d5a4e-b308-4625-92d5-4ca61399c83a.png">

**Comment by gregkalapos** [comment]
Hey @bmorelli25,

yeah, branches were not created, only the tags. I just created a branch based on the tag for the last release. So now we have the `1.17` branch.

**Comment by gregkalapos** [comment]
re: original question: huge thanks @bmorelli25 for the [comment above](https://github.com/elastic/apm-agent-dotnet/issues/1842#issuecomment-1260000096).

With that, it's now clear how this should work. We talked about this in our latest .NET syncs, and honestly no one was very convinced about how good this is compared to the way all other agents do it (which is to only have a single branch for each major version in the APM Agent repo and only link that).

I scheduled a meeting to discuss this - let's talk about this over zoom.
